otherlibs/win32unix/symlink.c

```diff
@@ -33,7 +33,10 @@ static int no_symlink = 0;
 CAMLprim value unix_symlink(value to_dir, value osource, value odest)
 {
   CAMLparam3(to_dir, osource, odest);
-  DWORD flags = (Bool_val(to_dir) ? SYMBOLIC_LINK_FLAG_DIRECTORY : 0);
+  DWORD flags =
+    Is_block(to_dir) && Bool_val(Field(to_dir, 0))
+    ? SYMBOLIC_LINK_FLAG_DIRECTORY
+    : 0;
```

Comment:
I'll can't do a full review today, but I can say immediately that this is wrong - see otherlibs/win32unix/unix.ml